src/wallet/walletdb.cpp

```diff
@@ -790,12 +793,15 @@ DBErrors CWalletDB::LoadWallet(CWallet* pwallet)
             string strType, strErr;
             if (!ReadKeyValue(pwallet, ssKey, ssValue, wss, strType, strErr))
             {
-                // losing keys is considered a catastrophic error, anything else
-                // we assume the user can live with:
-                if (IsKeyType(strType))
-                    result = DB_CORRUPT;
```

Comment:
A problem with removing this in favor of early return is that before this change, every key would be read into the in-memory wallet prior to returning control to the caller. By returning early, the in-memory wallet will be put into a partially-loaded state and so *must* not be used for anything if the result that's ultimately returned is not a critical error. So the caller must check the error state correctly. While this is something the caller *should* do, the type system can't guarantee that it's done (or that all the critical error states are correctly matched.) So @str4d suggests that this change be reverted and that the `result` variable be used as before instead of the early return, as a safeguard against any newly added critical error case being missed by the caller.

There should probably also be a comment to this effect added to this loop or to the `result` variable, to prevent this kind of change from accidentally being made again in the future. I would have made the same mistake. Damn you C++!